File: ruby/prime-factors/prime_factors.rb
class PrimeFactors

  def self.of(start)

    fac = []
    divisor = 2

    while start > 1
      while (start % divisor).zero?
        fac << divisor
        start /= divisor
      end

      divisor += 1
    end

    fac

  end

end

File: fsharp/prime-factors/PrimeFactors.fs
module PrimeFactors

let factors start =
    (start, 2L)
    |> List.unfold (function
        | 1L, _ -> None
        | n, divisor when n % divisor = 0L -> Some(divisor |> int |> Some, (n / divisor, divisor))
        | n, divisor -> Some(None, (n, divisor + 1L)))
    |> List.choose id
